src/lib.rs

```diff
@@ -63,9 +66,18 @@ const PROTOCOL_PERIOD: Duration = Duration::from_millis(1000);
 // TODO: Figure out what an actually optimal size would be.
 const INCOMING_BUFFER_SIZE: usize = 1024;
 
+/// Old recently a peer must have been put into the Known state for us to include them in the list
+/// of peers we send in response to KnownPeersRequest messages.
+const MAX_PEER_SHARE_AGE: Duration = Duration::from_millis(10000);
+
 /// How many other peers to ask to ping an unresponsive peer on our behalf.
 const NUM_INDIRECT_PING_PEERS: usize = 3;
 
+/// How many of our longest-since-heard-from peers to use as a potential candidate for our next
```

Comment:
Last week, I changed the "ping a random peer" logic to actually just ping whichever peer we haven't talked to for the longest time, which is an optimization from the SWIM paper that makes us guaranteed to discover downed nodes within a knowable time bound.

I was worried that this would lead to everyone who knows about Peer X getting into the "waiting for ping" state for the same peer in unison, though, so I added logic to pick the target peer from the oldest N, rather than the oldest 1.